### PR TITLE
feat: support health-check endpoints on hathor-core and tx-mining-service

### PR DESCRIPTION
**Pull request by luislhl**
### Acceptance Criteria
- Include methods to get health information on tx-mining-service and hathor-core

Note: The health endpoint in hathor-core is still being developed in https://github.com/HathorNetwork/hathor-core/pull/807. When it's released and available on most fullnodes, we will create a new PR to use it in the wallet-lib.

### Security Checklist
- [ ] Make sure you do not include new dependencies in the project unless strictly necessary and do not include dev-dependencies as production ones. More dependencies increase the possibility of one of them being hijacked and affecting us.

### CODE REVIEW

File: src/api/health.js [hunk 1]
```diff
@@ -0,0 +1,36 @@
...
+  /**
+   * Get health information of full node running in connected server
+   *
+   * @return {Promise}
+   * @memberof ApiHealth
+   * @inner
+   */
+  async getHealth() {
+    return new Promise((resolve, reject) => {
+      // TODO: We should chage this to get `health` instead of `version`
```
Comment (luislhl) [inline src/api/health.js:26]: This depends on https://github.com/HathorNetwork/hathor-core/pull/807

File: src/api/version.js [hunk 2]
```diff
@@ -29,6 +29,23 @@ const versionApi = {
...
+   *
+   * @return {Promise}
+   * @memberof ApiVersion
+   * @inner
+   */
+  async asyncGetVersion() {
+    return new Promise((resolve, reject) => {
+      createRequestInstance(resolve).get(`version`).then((res) => {
+        resolve(res.data);
+      }, (res) => {
```
Comment (tuliomir) [inline src/api/version.js:45]: :star2:  This wrapper is a welcome addition!

I would only suggest calling this object `err` , to reinforce it is not a regular result.
```suggestion
      }, (err) => {
```

File: src/api/health.js [hunk 3]
```diff
@@ -0,0 +1,36 @@
...
+      createRequestInstance(resolve).get(`version`).then((res) => {
+        resolve(res.data);
+      }, (res) => {
```
Comment (tuliomir) [inline src/api/health.js:29]: Same naming suggestion here, on the async error handler
```suggestion
      }, (err) => {
```

File: src/api/txMining.js [hunk 4]
```diff
@@ -73,6 +73,23 @@ const txMiningApi = {
...
+   *
+   * @return {Promise}
+   * @memberof ApiTxMining
+   * @inner
+   */
+  async getHealth() {
+    return new Promise((resolve, reject) => {
+      txMiningRequestClient(resolve).get(`health`).then((res) => {
+        resolve(res.data);
+      }, (res) => {
```
Comment (tuliomir) [inline src/api/txMining.js:88]: Also here
```suggestion
      }, (err) => {
```

**Review by tuliomir — COMMENTED** [review]
The code looks good to me, but it will be necessary to implement unit tests on those new methods for the CI to pass.

File: src/api/txMining.js (hunk 4 above)
Comment (luislhl) [reply src/api/txMining.js]: Done in https://github.com/HathorNetwork/hathor-wallet-lib/pull/579/commits/9951ac7161a6ea17df875f0a8e2d9eb258d4c99a

File: src/api/health.js (hunk 3 above)
Comment (luislhl) [reply src/api/health.js]: https://github.com/HathorNetwork/hathor-wallet-lib/pull/579/commits/9951ac7161a6ea17df875f0a8e2d9eb258d4c99a

File: src/api/version.js (hunk 2 above)
Comment (luislhl) [reply src/api/version.js]: Done in https://github.com/HathorNetwork/hathor-wallet-lib/pull/579/commits/9951ac7161a6ea17df875f0a8e2d9eb258d4c99a

File: __tests__/api/txMining.test.ts [hunk 5]
```diff
@@ -0,0 +1,77 @@
...
+
+    const result = await txMiningApi.getHealth();
+
+    expect(result).toEqual(data);
+  });
+
+  it('getHealth should allow capturing errors in case of network error', async () => {
+    mock.onGet('health').networkError();
+
+    await expect(txMiningApi.getHealth()).rejects.toThrow();
```
Comment (tuliomir) [inline __tests__/api/txMining.test.ts:56]: I like this test validating the error handling of `getHealth()`.

Should we have similar tests for `getJobStatus()`, `cancelJob()` and `submitJob()`?

Comment (luislhl) [reply __tests__/api/txMining.test.ts]: I've added some more tests in https://github.com/HathorNetwork/hathor-wallet-lib/pull/579/commits/ea887275286c208a12e434864712a8b2b2243a1a

**Review by tuliomir — APPROVED** [review]
:star2: Great organization for the unit tests.
Also, the async wrapper for `getVersion` will be quite useful for improving the wallet clients too.

File: src/api/health.js (hunk 1 above)
Comment (pedroferreira1) [reply src/api/health.js:26]: This was merged a week ago. Has it been released already?

**Review by pedroferreira1 — APPROVED** [review]

**Pull request merged**